guides/how-to-readme.md

```diff
@@ -0,0 +1,77 @@
+---
+explains: The goto file to see what a project is and how to use it
+---
+
+# How to write a `README`
+
+The `README` is the first thing almost anyone sees and perhaps one of the critical parts of the project. It serves as an advertisement for the product, indicator of for what users it can be relevant, the primary user’s guide and the way to get new collaborators up to speed.
+
+When writing the `README` keep your users in mind, they include:
+
+* Your teammates
+* Developers new to our organisation
+* (Potential) users of our projects
+* Contributors, both in and external
+* People interested in our work
+
+This guide goes into the primary elements any `README` should have, the bare minimum: what the project is; how to install, run and deploy; how to develop and test; and how to contribute.
+
+## 1.  The project name and description
+
+Give your project a short name by which people can refer to it. Preferably the repository name is similar. Descriptive names are great. However, feel free to get creative.
+
+The description is a high-level overview of what the project is for, what it does and in what way it should be used. The description can double as a mission to keep you focussed and on track, so it is wise to define this as early in the process as possible.
+
+> The `README` is an introduction. It should assume the reader knows absolutely nothing about the software and should provide a brief introduction. If software were a screenplay, the `README` would be the elevator pitch. If a person finishes reading the first 10 lines of `frobnicator/README` and still does not know if `frobnicator` is a widget library, accounting software, or a video game, then the author of the README has failed. — [William Pursell](https://stackoverflow.com/a/2305173)
+
+Be sure to answer these questions in your description:
+
+* What is this project?
+* How does it work?
+* Who are the users?
+* What is the benefit for the users?
+* What is the goal of this project?
+
+Start with the most important things so people can continue their search if this project is not what they are looking for.
+
+Make this section shine: If your project has a front-end include a screenshot here; if there is a demo or there are live examples, include links; if there are features, list them.
+
+## 2. How to install, configure, run and deploy
+
```

Comment:
This is the second most important part of the readme. The project should be as seflcontained as possible. Too many dependencies or installation steps scare off developers. So when you mention that you should document every step you're right, but I propose also to mention that the number of steps should be kept to a minumum.

And most important: test, test, test. If an installation fails a developer will quickly leave and search for another library

Reply:
I agree with all these points, however they are more points about how to make good software than about how to write a good `README`, as the readme should either be complete or 'point to relevant further docuementation if available'.

Reply:
Well, you could probably add a sentence to emphasize the importance of simple installation and configuration procedures

Reply:
I've added this in there as a tip, as it is mainly a project tip, more than a readme tip

`If this section gets long and tedious it might be worth it to try and simplify the process of getting started, as ease of use is one of the key reasons for developers to use a project.`